v1.5.0 fails to build on i686

```
Compiling mozim v0.1.0
error[E0277]: the trait bound `i32: From<u32>` is not satisfied
   --> /builddir/build/BUILD/netavark-1.5.0/vendor/mozim/src/socket.rs:289:17
    |
289 |         tv_sec: timeout.into(),
    |                 ^^^^^^^ ---- required by a bound introduced by this call
    |                 |
    |                 the trait `From<u32>` is not implemented for `i32`
    |
    = help: the following other types implement trait `From<T>`:
              <i32 as From<NonZeroI32>>
              <i32 as From<bool>>
              <i32 as From<i16>>
              <i32 as From<i8>>
              <i32 as From<nix::time::ClockId>>
              <i32 as From<nix::unistd::Pid>>
              <i32 as From<socket2::Domain>>
              <i32 as From<socket2::Protocol>>
            and 5 others
    = note: required for `u32` to implement `Into<i32>`
For more information about this error, try `rustc --explain E0277`.
error: could not compile `mozim` due to previous error
warning: build failed, waiting for other jobs to finish...
make: *** [Makefile:50: build] Error 101
error: Bad exit status from /var/tmp/rpm-tmp.sLYjfo (%build)
RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.sLYjfo (%build)
Child return code was: 1
```

Looks like an issue in a dependency crate, but that's blocking i686 rpm build.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -60,7 +60,7 @@ netlink-sys = "0.8.4"
 tokio = { version = "1.26", features = ["rt", "rt-multi-thread", "signal", "fs"] }
 tokio-stream = { version = "0.1", features = ["net"] }
 tonic = "0.8"
-mozim = "0.1"
+mozim = "0.2"
 prost = "0.11"
 futures-channel="0.3"
 futures-core = "0.3"

--- a/src/dhcp_proxy/dhcp_service.rs
+++ b/src/dhcp_proxy/dhcp_service.rs
@@ -45,7 +45,7 @@ pub enum DhcpClient {
 pub struct DhcpService {
     client: Option<DhcpClient>,
     network_config: NetworkConfig,
-    timeout: isize,
+    timeout: u32,
 }
 
 trait IP4Conv {
@@ -75,7 +75,7 @@ impl IP6Conv for IpAddr {
 }
 
 impl DhcpService {
-    pub fn new(nc: &NetworkConfig, timeout: isize) -> Result<DhcpService, DhcpServiceError> {
+    pub fn new(nc: &NetworkConfig, timeout: u32) -> Result<DhcpService, DhcpServiceError> {
         let client = Self::create_client(nc)?;
         Ok(DhcpService {
             client: Some(client),
@@ -183,10 +183,7 @@ impl DhcpService {
         match version {
             //V4
             0 => {
-                let config = match DhcpV4Config::new_proxy(iface, &nc.container_mac_addr) {
-                    Ok(c) => c,
-                    Err(e) => return Err(DhcpServiceError::new(InvalidArgument, e.to_string())),
-                };
+                let config = DhcpV4Config::new_proxy(iface, &nc.container_mac_addr);
                 match DhcpV4Client::init(config, None) {
                     Ok(client) => Ok(DhcpClient::V4Client(Box::new(client))),
                     Err(err) => Err(DhcpServiceError::new(InvalidArgument, err.to_string())),

--- a/src/dhcp_proxy/proxy_conf.rs
+++ b/src/dhcp_proxy/proxy_conf.rs
@@ -15,7 +15,7 @@ pub const DEFAULT_CONFIG_DIR: &str = "";
 // Default Network configuration path
 pub const DEFAULT_NETWORK_CONFIG: &str = "/dev/stdin";
 // Default epoll wait time before dhcp socket times out
-pub const DEFAULT_TIMEOUT: isize = 8;
+pub const DEFAULT_TIMEOUT: u32 = 8;
 // Proxy server gRPC socket file name
 pub const PROXY_SOCK_NAME: &str = "nv-proxy.sock";
 // Where leases are stored on the filesystem

--- a/src/dhcp_proxy_server/server.rs
+++ b/src/dhcp_proxy_server/server.rs
@@ -50,7 +50,7 @@ struct NetavarkProxyService<W: Write + Clear> {
     // cache is the lease hashmap
     cache: Arc<Mutex<LeaseCache<W>>>,
     // the timeout for the dora operation
-    dora_timeout: isize,
+    dora_timeout: u32,
     // channel send-side for resetting the inactivity timeout
     timeout_sender: Arc<Mutex<Sender<i32>>>,
 }
@@ -186,7 +186,7 @@ struct Opts {
     uds: Option<String>,
     /// optional time in seconds to time out after looking for a lease
     #[clap(short, long)]
-    timeout: Option<isize>,
+    timeout: Option<u32>,
     /// activity timeout
     #[clap(short, long)]
     activity_timout: Option<u64>,

